### PR TITLE
Making common subclass for Bigtable DirectRow and ConditionalRow.

### PR DESCRIPTION
**Pull request by dhermes**
This is a follow-up to #1557 and prevents strange-ly true statements like `isinstance(cond_row, DirectRow)`.

Sorry it took me so many days. Code has been completely written for 2+ days but I haven't ever had my laptop near internet long enough to push it.

### CODE REVIEW

File: docs/bigtable-row.rst [hunk 1]
```diff
@@ -63,5 +63,6 @@ level. For example:
   :members:
   :undoc-members:
   :show-inheritance:
+  :inherited-members:
```
Comment (dhermes) [inline docs/bigtable-row.rst]: This was for `ALL_COLUMNS` to show up even though it is on a non-public base class.

File: scripts/run_pylint.py [hunk 2]
```diff
@@ -66,7 +66,7 @@
 }
 TEST_RC_REPLACEMENTS = {
     'FORMAT': {
-        'max-module-lines': 1825,
+        'max-module-lines': 1840,
```
Comment (dhermes) [inline scripts/run_pylint.py]: Sorry. I really need to tear the filters out of `gcloud.bigtable.row` and give them their own home.

Comment (tseaver) [inline scripts/run_pylint.py]: Why should we be fiddling these majyk numbers just to get pylint to pass, if they don't represent "real" opinions?  https://github.com/tseaver/gcloud-python/commit/a43640dfb88db0955aac00f1614f2dff00cb7fa7 just turns off `too-many-lines` (and `too-many-statements`) for test code altogether, which I find preferable.

File: docs/bigtable-row.rst (hunk 1 above)
Comment (tseaver) [inline docs/bigtable-row.rst]: SGTM

**Comment by tseaver** [comment]
LGTM

File: scripts/run_pylint.py (hunk 2 above)
Comment (dhermes) [inline scripts/run_pylint.py]: I'll just send the PR to pull out all the filters.

Comment (dhermes) [inline scripts/run_pylint.py]: Sent #1602. Will update this after that goes in.

**Comment by dhermes** [comment]
@tseaver I rebased after #1602. Will merge once Travis passes.

**Pull request merged**